tests/apollo/test_skvbc_backup_restore.py

```diff
@@ -101,19 +106,198 @@ async def test_checkpoint_propagation_after_restarting_replicas(self, bft_networ
+        """
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+
+        n = bft_network.config.n
+
+        self.assertEqual(len(bft_network.procs), n, "Make sure all replicas are up initially.")
+
+        current_primary = await bft_network.get_current_primary()
+
+        initial_view = 0
+
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=current_primary)
+
+        await skvbc.fill_and_wait_for_checkpoint(
+            initial_nodes=bft_network.all_replicas(),
+            num_of_checkpoints_to_add=1,
+            verify_checkpoint_persistency=False
+        )
+
+        checkpoint_after = await bft_network.wait_for_checkpoint(replica_id=current_primary)
+        self.assertEqual(checkpoint_before + 1, checkpoint_after)
+
+        # stop n replicas in a random order with a delay of 10/n seconds in between, so that
+        # all replicas are stopped by 10seconds, and no view change is triggered during stopping
+        # of replicas.
+        stopped_replicas = await self._stop_random_replicas_with_delay(bft_network, delay=10/n)
+
+        stopped_replicas.remove(current_primary)
+
+        # start stopped replicas in a random order with a delay of 10s in between
+        # view change only happens if the initial primary starts at time > viewchangeTimer
+        # to make the test robust we start primary at the end.
+        await self._start_random_replicas_with_delay(bft_network, stopped_replicas, current_primary, delay=10)
+
+        # wait for view to stabilize
+        for replica in bft_network.all_replicas():
+            if await self._wait_for_view_under_constant_load(
+                    replica_id=replica,
+                    bft_network=bft_network,
+                    expected=lambda v: v > initial_view,
+                    err_msg="Make sure view is stable after all replicas are started."):
+                break
+
+        current_primary = await bft_network.get_current_primary()
+
+        await self._wait_for_processing_window_after_view_change(current_primary, bft_network)
+
+        # verify checkpoint persistency
+        await bft_network.wait_for_replicas_to_checkpoint(bft_network.all_replicas(),
+                                                          expected_checkpoint_num=lambda ecn: ecn >= checkpoint_after)
+
+        await self._fill_and_wait_for_checkpoint_under_constant_load(skvbc,
+                                                                     bft_network,
+                                                                     initial_nodes=bft_network.all_replicas(),
+                                                                     num_of_checkpoints_to_add=1,
+                                                                     verify_checkpoint_persistency=False,
+                                                                     assert_state_transfer_not_started=False)
```

Comment:
I suggest we get the checkpoint_before and the checkpoint_after and assert that checkpoint_after >= checkpoint_before.